server/game/cards/09.1-FtE/MomentOfPerfectBeauty.js

```diff
@@ -0,0 +1,23 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl');
+const { Players, Durations } = require('../../Constants');
+
+class MomentOfPerfectBeauty extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'One more action and then end the conflict',
+            condition: context => this.game.isDuringConflict() &&
+                    this.game.currentConflict.getNumberOfParticipantsFor(context.player, card => card.isHonored) >
+                    this.game.currentConflict.getNumberOfParticipantsFor(context.player.opponent, card => card.isHonored),
+            gameAction: AbilityDsl.actions.playerLastingEffect({
+                duration: Durations.UntilEndOfConflict,
+                targetController: Players.Opponent,
+                effect: AbilityDsl.effects.resolveConflictEarly()
+            })
+        });
```

Comment:
You could do with an `effect:` here otherwise you will not get a very meaningful chat message.  Something like:

`effect: 'resolve the conflict after {1]'s next action',
`effectArgs`: context => [context.player.opponent]